packages/nimble-components/src/rich-text-viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -10,4 +29,209 @@ describe('RichTextViewer', () => {
     it('should export its tag', () => {
         expect(richTextViewerTag).toBe('nimble-rich-text-viewer');
     });
+
+    it('should reflect value of the "markdownValue" to the internal control', async () => {
+        await connect();
+
+        const markdownValue = '**markdown string**';
+        element.setAttribute('markdownValue', markdownValue);
+
+        expect(element.getAttribute('markdownValue')).toBe(markdownValue);
+
+        await disconnect();
+    });
+
+    describe('supported rich text formatting options from markdown string to its respective HTML elements', () => {
+        let childElement: Element | null | undefined;
+        function expectChildElements(
+            childElementTagNames: string[],
+            expectedText: string
+        ): void {
+            childElement = pageObject.getFirstChildElement();
+            for (const childTagName of childElementTagNames) {
+                expect(pageObject.getTagName(childElement)).toBe(childTagName);
+                if (
+                    childElementTagNames.indexOf(childTagName)
+                    !== childElementTagNames.length - 1
+                ) {
+                    childElement = childElement?.firstElementChild;
+                }
+            }
+            expect(pageObject.getTextContent(childElement)).toBe(expectedText);
+        }
+
+        it('should convert bold markdown string to "strong" HTML tag', async () => {
+            element.markdownValue = '**Bold**';
+
+            await connect();
+
+            expectChildElements(['P', 'STRONG'], 'Bold');
```

Comment:
- Conceptually avoid placing expect statements inside these helper functions. Instead the helper functions should return values that you can do the expect on.
- In addition, avoid closing around shared state in the helper functions. That makes them difficult to share or re-use in the future. Instead you should pass state they need as parameters.

For example
```ts
// nect to other helpers like the setup() function
function getChildElements(pageObject): string[] {}

...

it('something', async () => {
  element.markdownValue = '**Bold**';
  await connect();
  
  expect(getChildElements(pageObject)).toEqual(['P', 'STRONG']);
  // this one doesn't look like it even needs a helper
  expect(pageObject.getTextContent(childElement)).toBe(expectedText);
});
```

This type of organization keeps the logic of gathering the test results and evaluating the expect clearly separated.

Reply:
I have updated the helper function to get the element tag names instead of expecting it there.

Reply:
Re-opening, the second bullet was not addressed.
Pass all needed state as parameters instead of closing around running test state.

Reply:
My concern was the state being [closed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) by the helper.

Compare:
```
function outer() {
   let pageObject;

  function helper() {
    console.log(pageobject)
  }
}
```
That function is closed around a variable in the parent scope.

Compare to:
```
function outer() {
   let pageObject;

  function helper(pageObject) {
    console.log(pageobject)
  }
}
```
That helper function is no longer closed around specific state. It takes state as parameters.

By moving to the page object you effectively resolved my concern of helpers being closed around variables in the test. My concern is closing around test state in helper functions instead of passing as a parameters. But the current state of nicely formatted page object methods is even better! ✨